test/modules/TestResultCoordinator/Reports/DirectMethod/LongHaul/DirectMethodLongHaulReportGenerator.cs

```diff
@@ -55,15 +56,16 @@ class DirectMethodLongHaulReportGenerator : ITestResultReportGenerator
 
         public async Task<ITestResultReport> CreateReportAsync()
         {
-            ulong senderSuccesses = 0;
-            Option<ulong> receiverSuccesses = Option.None<ulong>();
-            ulong statusCodeZero = 0;
-            ulong unknown = 0;
+            long senderSuccesses = 0;
+            Option<long> receiverSuccesses = Option.None<long>();
+            long statusCodeZero = 0;
+            Dictionary<int, long> other = new Dictionary<int, long>();
             while (await this.SenderTestResults.MoveNextAsync())
             {
                 this.ValidateDataSource(this.SenderTestResults.Current, this.SenderSource);
                 DirectMethodTestResult dmSenderTestResult = JsonConvert.DeserializeObject<DirectMethodTestResult>(this.SenderTestResults.Current.Result);
-                switch ((int)dmSenderTestResult.HttpStatusCode)
+                int statusCode = (int)dmSenderTestResult.HttpStatusCode;
```

Comment:
Hmm.... I have a mixed feeling about this. One hand, you can just have the dict with key type `<System.Net.HttpStatusCode ,long>` or you could have `<int, long>`. Personally, I'd prefer to have `<System.Net.HttpStatusCode ,long>` as it carries out how to interpret the key through the code. I'll leave it up to you to decide.